File: realsense-test.cpp
#include <librealsense2/rs.hpp>
#include <iostream>

#define WIDTH 640
#define HEIGHT 480
#define FPS 30

int main() {
    try {
        rs2::pipeline pipe;
        rs2::config cfg;

        cfg.enable_stream(RS2_STREAM_COLOR, WIDTH, HEIGHT, RS2_FORMAT_BGR8, FPS);
        cfg.enable_stream(RS2_STREAM_DEPTH, WIDTH, HEIGHT, RS2_FORMAT_Z16, FPS);

        auto profile = pipe.start(cfg);

        rs2::align align_to_color(RS2_STREAM_COLOR);

        while (true) {
            rs2::frameset fs = pipe.wait_for_frames();
            fs = align_to_color.process(fs);
            rs2::depth_frame d = fs.get_depth_frame();
            float dist = d.get_distance(320, 240);
            std::cout << "\rCenter distance: " << dist << " m" << std::flush;
        }
   } catch (const rs2::error& e) {
        std::cerr << "RealSense error: " << e.what() << "\n";
        return 1;
   } catch (const std::exception& e) {
        std::cerr << "Std error: " << e.what() << "\n";
        return 2;
   }
}

File: realsense-test.py
import pyrealsense2 as rs
import numpy as np
import cv2

W, H, FPS = 640, 480, 60
pipe = rs.pipeline()
config = rs.config()
config.enable_stream(rs.stream.color, W, H, rs.format.bgr8, FPS)
config.enable_stream(rs.stream.depth, W, H, rs.format.z16, FPS)

profile = pipe.start(config)
align = rs.align(rs.stream.color)
depth_scale = profile.get_device().first_depth_sensor().get_depth_scale()

try:
    while True:
        frames = pipe.wait_for_frames()
        frames = align.process(frames)
        c = frames.get_color_frame()
        d = frames.get_depth_frame()
        if not c or not d:
            continue
        color = np.asanyarray(c.get_data())
        depth = np.asanyarray(d.get_data())

        dist_center = depth[H // 2, W // 2] * depth_scale
        cv2.putText(color, f"{dist_center}", (10, 30), cv2.FONT_HERSHEY_SIMPLEX, 1, (0, 255, 0), 2)
        cv2.imshow("Color", color)
        if cv2.waitKey(1) == 27:
            break
finally:
    pipe.stop()
    cv2.destroyAllWindows()
    print("Pipeline stopped and windows closed.")
    print("Depth scale:", depth_scale)
    print("Resolution:", W, "x", H)
    print("FPS:", FPS)
    print("Test completed successfully.")
    print("Exiting...")
    exit(0)